### PR TITLE
Add support for header parameters in service store

### PR DESCRIPTION
**Pull request by hardikmaheshwari**
## Summary
Add support for header parameters in service store.
Ref - https://swagger.io/docs/specification/describing-parameters/

## How did you test this change?

- [X] Test(s) added
- [ ] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

**Comment by akphi** [comment]
@hardikmaheshwari it's filing engine test, let's not merge till we get engine through

**Comment by akphi** [comment]
I converted this to draft as the corresponding engine's PR is in draft mode and not yet merged. Please feel free to switch back when engine change is in! Thanks!

**Comment by hardikmaheshwari** [comment]
> I converted this to draft as the corresponding engine's PR is in draft mode and not yet merged. Please feel free to switch back when engine change is in! Thanks!

@akphi Engine changes were merged as part of another PR. Can you please review this?

File: .changeset/fresh-coats-roll.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@finos/legend-extension-external-store-service": patch
```
Comment (akphi) [inline .changeset/fresh-coats-roll.md:2]:
```suggestion
"@finos/legend-extension-external-store-service": minor
```

**Pull request merged**